task/power/src/main.rs

```diff
@@ -568,6 +587,32 @@ impl idl::InOrderPowerImpl for ServerImpl {
         Ok(out.0)
     }
 
+    fn phase_current(
+        &mut self,
+        _msg: &userlib::RecvMessage,
+        rail: task_sensor_api::SensorId,
```

Comment:
Vague thoughts: using the voltage sensor as a unique ID for a rail is interesting, and I _think_ I like it.

I ran into the issue of identifying a RAA/ISL part when adding the blackbox functions, and decided on using their I2C address (which happen to be unique); using the `SensorId` is more _obviously_ unique.

However, a `SensorId` could be many-to-one if there are multiple voltage sensors associated with a single rail, and it's _definitely_ many-to-one if we used it to identify a RAA/ISL part as a whole (instead of a single rail).

Reply:
Thinking about this a little more, I'd like to propose an alternative: how about a Hubris endpoint that goes from a `SensorId` to a `usize`, which is the index in `CONTROLLER_CONFIG`?

This would mean that we could do the lookup _once_, then use that index repeatedly.  I'm thinking ahead to the per-phase code that I'll be writing soon, which requires a _bunch_ of device reads and writes; it would be nice to not have an O(N) loop to look up the device each time.

Alternatively, we could make `CONTROLLER_CONFIG` into a `static` rather than `const`, and have *Humility* find the index.  This moves the iteration logic into the Big Computer, although it couples us more tightly to the details of layout – we'd basically be scanning the list below for a matching `SensorId`.

```
matt@niles ~ () $ h -tgimlet readvar CONTROLLER_CONFIG
humility: attached to 0483:374f:000C001F4D46500F20373033 via ST-Link V3
task_power::bsp::CONTROLLER_CONFIG (0x08014aa4) = [
    PowerControllerConfig {
        state: A2,
        device: IBC,
        builder: 0x80121e9 (fn(abi::TaskId) -> (drv_i2c_api::I2cDevice, u8)),
        voltage: SensorId(0x74),
        input_voltage: None,
        current: SensorId(0x73),
        input_current: None,
        temperature: Some(SensorId(0x71))
    },
    PowerControllerConfig {
        state: A0,
        device: Core,
        builder: 0x8012243 (fn(abi::TaskId) -> (drv_i2c_api::I2cDevice, u8)),
        voltage: SensorId(0x4f),
        input_voltage: None,
        current: SensorId(0x4d),
        input_current: None,
        temperature: Some(SensorId(0x49))
    },
    PowerControllerConfig {
        state: A0,
        device: Core,
        builder: 0x80122af (fn(abi::TaskId) -> (drv_i2c_api::I2cDevice, u8)),
        voltage: SensorId(0x57),
        input_voltage: None,
        current: SensorId(0x55),
        input_current: None,
        temperature: Some(SensorId(0x51))
    },
    ...
```

Reply:
Yes, agreed with all of this -- and I liked this better than the alternatives, though it is not without its own issues.